packages/backend-common/src/reading/git.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import isomorphicGit, { ProgressCallback, AuthCallback } from 'isomorphic-git';
+import http from 'isomorphic-git/http/node';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { isAuthCallbackOptions } from './GerritUrlReader';
```

Comment:
Prolly best to flip this around right? i.e. move it to this file and have the `GerritUrlReader` import instead.

Reply:
Good catch, fixed 🙏🏻